syntax = "proto3";

// Not necessary for Python but should still be declared to avoid name collisions 
// in the Protocol Buffers namespace and non-Python languages
package cctv;


message Frame {
    message RawFrame {
        string movie_filename = 1;
        string movie_filepath = 2;
        bytes image_bytes = 3;
    }

    RawFrame raw_frame = 1;
    float t_created = 2;
    float t_updated = 3;

    repeated string objects = 4;
    repeated bytes faces = 5;
    repeated string matched_faces = 6;
    
}

